jar-infer/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -60,27 +58,34 @@
     AnalysisCache cache = new AnalysisCacheImpl();
     IClassHierarchy cha = ClassHierarchyFactory.make(scope);
     Warnings.clear();
+    HashMap<String, Set<String>> result = new HashMap<String, Set<String>>();
 
-    MethodReference method =
-        scope.findMethod(
-            AnalysisScope.APPLICATION,
-            mainClass,
-            Atom.findOrCreateUnicodeAtom(targetMethod),
-            new ImmutableByteArray(UTF8Convert.toUTF8(targetMethodSignature)));
-    Preconditions.checkNotNull(method, "method not found");
-    IMethod imethod = cha.resolveMethod(method);
-    Preconditions.checkNotNull(imethod, "imethod not found");
-    IR ir = cache.getIRFactory().makeIR(imethod, Everywhere.EVERYWHERE, options.getSSAOptions());
-    System.out.println(ir);
-
-    ControlFlowGraph<SSAInstruction, ISSABasicBlock> cfg = ir.getControlFlowGraph();
-    DefinitelyDerefedParams derefedParamsFinder =
-        new DefinitelyDerefedParams(imethod, ir, cfg, cha);
-    Set<String> result = derefedParamsFinder.analyze();
-
+    // Iterate over all classes:methods in the 'Application' and 'Extension' class loaders
```

Comment:
One question here: What happens if I have a library `A.jar`, which depends on say `B.jar`, but I don't necessarily wish to infer nullability for `B` at this time, just `A`, will that work in this case? If not, what will happen, will WALA fail because we never explicitly added `B.jar`? Or will it get analyzed no matter what?

This can also be a concern when we have a library like `guava` or `rxjava`, where a bunch of other libraries depend on it and we don't want its classes to be analyzed multiple times. Basically, we need to make sure this code iterates over all classes in the Jar and none outside of it.